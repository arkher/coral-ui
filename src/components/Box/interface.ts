import { BoxProps } from '@shopify/restyle';
import { ReactNode } from 'markdown-to-jsx/node_modules/@types/react';
import { Theme } from '../../themes/default';

type CustomBoxProps = Partial<{
  children: ReactNode;
  bw: CustomBorderWidth;
  op: CustomOpacity;
  sof: CustomShadow;
  sr: CustomShadow;
  sop: CustomShadow;
  h: CustomHeightComponent;
}> &
  Partial<BoxProps<Theme>>;

export { CustomBoxProps };
