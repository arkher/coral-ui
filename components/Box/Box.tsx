import { BoxProps, createBox, createRestyleComponent } from '@shopify/restyle';
import React, { ReactNode } from 'react';
import {
  borderWidth,
  opacity,
  shadowOffset,
  shadowOpacity,
  shadowRadius,
  Theme,
} from '../../themes/default';

const Box = createBox<Theme>();

type CoralBoxProps = Partial<{
  children?: ReactNode;
  bw?: CustomBorderWidth;
  op?: CustomOpacity;
  sof?: CustomShadow;
  sr?: CustomShadow;
  sop?: CustomShadow;
}> &
  Partial<BoxProps<Theme>>;

const CoralBox: React.FC<CoralBoxProps> = ({ children, ...props }) => (
  <Box {...props}>{children}</Box>
);

export default createRestyleComponent<CoralBoxProps, Theme>(
  [borderWidth, opacity, shadowOffset, shadowRadius, shadowOpacity],
  CoralBox,
);
