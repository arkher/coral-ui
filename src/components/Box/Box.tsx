import { createBox, createRestyleComponent } from '@shopify/restyle';
import React from 'react';
import {
  borderWidth,
  opacity,
  shadowOffset,
  shadowOpacity,
  shadowRadius,
  Theme,
  heightComponent,
} from '../../themes/default';
import { CustomBoxProps } from './interface';

const Box = createBox<Theme>();

const CustomBox: React.FC<CustomBoxProps> = ({ children, ...props }) => (
  <Box {...props} testID="Box">
    {children}
  </Box>
);

export default createRestyleComponent<CustomBoxProps, Theme>(
  [
    borderWidth,
    opacity,
    shadowOffset,
    shadowRadius,
    shadowOpacity,
    heightComponent,
  ],
  CustomBox,
);
