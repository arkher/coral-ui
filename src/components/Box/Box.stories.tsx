/* eslint-disable arrow-body-style */
import React from 'react';
import { View } from 'react-native';
import { COLORS, RADIUS, OPACITIES, SHADOWS } from '../../themes/tokens';
import Box from './Box';

export default {
  title: 'Elements/Box',
  component: Box,
  argTypes: {
    bg: {
      options: COLORS,
      control: { type: 'select' },
    },
    borderRadius: {
      options: RADIUS,
      control: { type: 'select' },
    },
    opacity: {
      options: OPACITIES,
      control: { type: 'select' },
    },
    shadow: {
      options: SHADOWS,
      control: { type: 'select' },
    },
    width: {
      control: { type: 'range', min: 100, max: 500, step: 100 },
    },
    height: {
      control: { type: 'range', min: 100, max: 500, step: 100 },
    },
  },
};

interface BoxProps {
  bg: Custom.Colors;
  br: Custom.Radius;
  opacity: Custom.Opacity;
  shadow: Custom.Shadow;
  width: number;
  height: number;
}

export const Default = ({
  bg,
  br,
  opacity,
  shadow,
  width,
  height,
}: BoxProps): React.ReactNode => {
  return (
    <View style={{ margin: 8 }}>
      <Box
        bg={bg || 'primary-base'}
        br={br}
        op={opacity}
        sof={shadow}
        sr={shadow}
        sop={shadow}
        width={width || 100}
        height={height || 50}
      />
    </View>
  );
};
