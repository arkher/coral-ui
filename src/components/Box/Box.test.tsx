import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import Box from './Box';
afterEach(cleanup);
test('renderiza corretamente', async () => {
  const { toJSON } = render(<Box />);

  expect(toJSON()).toMatchSnapshot();
});
