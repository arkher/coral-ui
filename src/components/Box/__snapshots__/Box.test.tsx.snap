// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza corretamente 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
  testID="Box"
/>
`;
